catalog/PartitionSchemeHeader.cpp

```diff
@@ -32,16 +34,29 @@
 
 #include "glog/logging.h"
 
+using std::move;
+using std::size_t;
+using std::vector;
+
 namespace quickstep {
 
 PartitionSchemeHeader::PartitionSchemeHeader(const PartitionType type,
                                              const std::size_t num_partitions,
-                                             const attribute_id attr_id)
+                                             PartitionAttributeIds &&attr_ids)  // NOLINT(whitespace/operators)
     : partition_type_(type),
       num_partitions_(num_partitions),
-      partition_attribute_id_(attr_id) {
+      partition_attribute_ids_(move(attr_ids)) {
   DCHECK_GT(num_partitions, 0u);
-  DCHECK_GE(attr_id, 0);
+
+#ifdef QUICKSTEP_DEBUG
```

Comment:
Optional, but a couple of lines regarding what you are trying to do here should be helpful.